README.md

```diff
@@ -1,81 +1,167 @@
+
+- Descriptor and reflection support
+- ECMAScript module support
+- First-class TypeScript support
+- Generation of idiomatic JavaScript and TypeScript code.
+- Generation of [much smaller bundles](packages/protobuf-bench)
+- Implementation of all proto3 features, including the [canonical JSON format](https://developers.google.com/protocol-buffers/docs/proto3#json).
+- Implementation of all proto2 features, except for extensions and the text format.  
+- Usage of standard JavaScript APIs instead of the [Closure Library](http://googlecode.blogspot.com/2009/11/introducing-closure-tools.html)
+- Compatibility is covered by the protocol buffers [conformance tests](packages/protobuf-conformance).
+
+Perhaps the strongest feature of **Protobuf-ES** mentioned above is its generation of idiomatic JavaScript and TypeScript code.  It adopts the best features from the community generators. This means no more clunky getters and setters. You can now use things like the spread operator and make use of the same JavaScript semantics you've grown used to.
+
+For example, given a Protobuf file such as:
+
+```protobuf
+syntax="proto3";
+
+message User {
+  string first_name = 1;
+  string last_name = 2;
+  bool active = 3;
+  User manager = 4;
+  repeated string locations = 5;
+  map<string, string> projects = 6;
+}
+```
+
+You can initialize your objects conveniently using the `new` operator or passing an initializer object to constructors:
+
+```typescript
+// Using new
+let user = new User();
+
+// Using an object in the constructor
+user = new User({
+  firstName: "Homer",
+  lastName: "Simpson",
+  manager: {  // you can simply pass an initializer object for this nested message field
+      firstName: "Montgomery",
+      lastName: "Burns"
+  },
+});
+```
+
+In addition, you can use direct property access and work with the same familiar JavaScript syntax you're used to.  For example:
+
+```typescript
+user.active = true;
+user.manager.firstName = "Monty";
+msg.locations = ["Springfield"];
+msg.projects = {
+   "SPP": "Springfield Power Plant"
+};
+```
+
+
 To learn more, have a look at a complete [code example](https://github.com/bufbuild/protobuf-es/tree/main/packages/protobuf-example), 
 the documentation for the [generated code](https://github.com/bufbuild/protobuf-es/blob/main/docs/generated_code.md), 
 and the documentation for the [runtime API](https://github.com/bufbuild/protobuf-es/blob/main/docs/runtime_api.md).
 
+## Getting Started
 
-### How does this compare to protoc's JavaScript generator?
+Installation instructions vary depending on your purpose.  The instructions for npm are shown below but Yarn and pnpm are also supported.
 
-[`js_generator.cc`](https://github.com/protocolbuffers/protobuf-javascript/blob/main/generator/js_generator.cc)
-is rarely updated, and has fallen behind the quickly moving world of JavaScript.
+### [@bufbuild/protobuf](https://www.npmjs.com/package/@bufbuild/protobuf)
```

Comment:
You could omit this section entirely. I think the next section is the relevant part.